drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -294,6 +309,42 @@ local function temp_attr_handler(driver, device, ib, response)
   end
 end
 
+local mired_bounds_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value ~= nil then
+      if (ib.data.value < COLOR_TEMPERATURE_MIRED_MIN or ib.data.value > COLOR_TEMPERATURE_MIRED_MAX) then
+        device.log.warn_with({hub_logs = true}, string.format("Device reported a min or max color temperature %d mired outside of supported capability range", ib.data.value))
+        return
+      end
+      local temp_in_kelvin = mired_to_kelvin(ib.data.value)
+      device:set_field(COLOR_TEMP_BOUND_RECEIVED..minOrMax, temp_in_kelvin)
+      local min = device:get_field(COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MIN)
+      local max = device:get_field(COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MAX)
+      if min ~= nil and max ~= nil and min < max then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorTemperature.colorTemperatureRange({ value = {minimum = min, maximum = max} }))
+        device:set_field(COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MAX, nil)
+        device:set_field(COLOR_TEMP_BOUND_RECEIVED..COLOR_TEMP_MIN, nil)
+      end
+    end
+  end
+end
+
+local level_bounds_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
```

Comment:
This is fine, but I would argue that you can still be DRY by putting everything below the the call to `set_field` into a common function, and keeping the handlers for min and max split, which would be more readable (IMO).

Reply:
In order to stay aligned with the Zigbee implementation and the other matter drivers, I would prefer to leave this as-is for now.